File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
// 최대공약수 찾기
function findGcd(a, b) {
    return b === 0 ? a : findGcd(b, a % b)
}

function solution(numer1, denom1, numer2, denom2) {
    const sumDenom = denom1 * denom2;
    const sumNumer = numer1 * denom2 + numer2 * denom1;
    const gcd = findGcd(sumDenom, sumNumer);
    return [sumNumer/gcd, sumDenom/gcd];
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
import math

def solution(numer1, denom1, numer2, denom2):
    sum_denom = denom1 * denom2
    sum_numer = numer1 * denom2 + numer2 * denom1
    gcd = math.gcd(sum_denom, sum_numer)
    return [sum_numer/gcd, sum_denom/gcd]